src/dmd/dtoh.d

```diff
@@ -2029,8 +2030,30 @@ public:
             scope(exit) printf("[AST.RealExp exit] %s\n", e.toChars());
         }
 
-        // TODO: Needs to implemented, properly switching on the e.type
-        buf.printf("%ff", cast(double)e.value);
+        import core.stdc.math;
+        if (isnan(e.value))
+        {
+            buf.writestring("NAN");
+        }
+        else if (isinf(e.value))
+        {
+            buf.writestring("INFINITY");
+        }
+        else switch (e.type.ty) with (AST.ENUMTY)
+        {
+            case Tfloat32:
+                buf.printf("%ff", cast(double) e.value);
+                break;
+            case Tfloat64:
+                buf.printf("%f", cast(double) e.value);
+                break;
+            case Tfloat80:
+                buf.printf("%Lfl", e.value);
+                break;
+            default:
+                printf("%s = %d\n", e.type.toChars(), e.type.ty);
+                assert(false);
+        }
```

Comment:
Thanks! - Please use the `dmd.root.ctfloat.CTFloat.{isNaN,isInfinity,sprint}` functions to check and print `real_t` values (not necessarily corresponding to the host C `long double`, not just for non-DMD compilers, but also for DMD linked against MSVC). I'd also prefer an uppercase `L` as `long double` suffix. And detecting `-inf` might be required as well.

Reply:
The printing logic, incl. falling back to a hex string if the compact decimal representation isn't accurate enough, for non-Nan/inf is already in [hdrgen](https://github.com/dlang/dmd/blob/0470e5988f29fc278cb5cb86a3e42300a8b4d9ce/src/dmd/hdrgen.d#L1883) and could be refactored.

Reply:
Agreed, gdc doesn't use native floating point (either?), so casting to float/double is futile. ;-)

Reply:
Agreed, gdc doesn't use native floating point (either?), so casting to float/double is futile. ;-)

Reply:
Thanks! Changed the implementation to use `CTFloat` and `floatToBuffer`.